### DISCUSSION TITLE
Naming support for the BED format

### DISCUSSION THREAD

**Discussion by smehringer** [biocpp/biocpp-io#51] [Design Questions]
from thread in https://github.com/h-2/b.i.o./pull/48

How to name the support for annotation file types (BED, bedGraph, wiggle, etc.).

> Regarding the name: I still think that "ann_io" is sounds like the name of a secret agent 🕵🏻‍♀️ but I also don't have a great replacement, so we can stick with this for now. We can also discuss with @smehringer to see what she thinks.

Questions: which formats will be reflected? E.g. what about GTF?

**Comment by smehringer** [comment]
Note in general: In seqan3 we started out to combine multiple formats (e.g. alignment IO combining SAM and blast) while in seqan2 they were separate. B.I.O now mimics seqan2 rather than seqan3 (map_io will most probably not contain blast io).

For me the decision depends highly on which formats will be included.
Possibilities are `ann_io`, `bed_io` for a start, `region_io` (I encountered BED files merely as annotation a region, if we call it `ann_io` a vcf file could also be seen as an annotation file, why is that not included)

**Comment by joshuak94** [comment]
I would say GTF could also be supported, but my plan was to for now just implement BED/bedGraph/WIG files.

That's a very good point that VCF files could be considered a part of annotation IO. Another option could be `track_io`, for formats meant to be displayed on a track (genome browser, UCSC track, etc.) taken from UCSC FAQ: "BED (Browser Extensible Data) format provides a flexible way to define the data lines that are displayed in an annotation track."

I would be fine with `region_io` or `track_io`, I just thought we wanted to stick with three letters (`map_io` and `vcf_io` and `seq_io`)
